java/src/main/java/ai/rapids/cudf/Rmm.java

```diff
@@ -198,18 +204,21 @@ public static synchronized void initialize(int allocationMode, LogConf logConf,
    * @throws IllegalStateException if RMM has already been initialized
    * @throws IllegalArgumentException if a max pool size is specified but the allocation mode
    *                                  is not {@link RmmAllocationMode#POOL} or
-   *                                  {@link RmmAllocationMode#ARENA}, or the maximum pool size is
-   *                                  below the initial size.
+   *                                  {@link RmmAllocationMode#ARENA} or
+   *                                  {@link RmmAllocationMode#CUDA_ASYNC}, or the maximum pool
+   *                                  size is below the initial size.
    */
   public static synchronized void initialize(int allocationMode, LogConf logConf, long poolSize,
       long maxPoolSize, long allocationAlignment, long alignmentThreshold) throws RmmException {
     if (initialized) {
       throw new IllegalStateException("RMM is already initialized");
     }
     if (maxPoolSize > 0) {
-      if (allocationMode != RmmAllocationMode.POOL && allocationMode != RmmAllocationMode.ARENA) {
+      if (allocationMode != RmmAllocationMode.POOL &&
+          allocationMode != RmmAllocationMode.ARENA &&
+          allocationMode != RmmAllocationMode.CUDA_ASYNC) {
```

Comment:
Not a new bug, but allocationMode values were designed as individual bits so they could be combined together.  By checking the raw values here, we can't configure a pooling allocator like POOL or ARENA with managed memory underneath (e.g.: configure a 32GB max pool size on a 16GB GPU using managed memory to handle the overflow on the device).

Reply:
Fixed.